src/Date.php

```diff
@@ -103,23 +103,30 @@ public static function createFromFormat($format, $time, $timezone = null)
      * Alias for diffForHumans.
      *
      * @param  Date $since
-     * @param  bool $absolute Removes time difference modifiers ago, after, etc
+     * @param  bool $syntax  Removes time difference modifiers ago, after, etc
+     * @param  bool $short   (Carbon 2 only) displays short format of time units
+     * @param  int  $parts   (Carbon 2 only) maximum number of parts to display (default value: 1: single unit)
+     * @param  int  $options (Carbon 2 only) human diff options
      * @return string
      */
-    public function ago($since = null, $absolute = false)
+    public function ago($since = null, $syntax = null, $short = false, $parts = 1, $options = null)
```

Comment:
Since this method is `ago` we can probably still keep the old method format right? So just `since` and `absolute`?

Reply:
In Carbon 2, $syntax = $absolute + relative to now/relative to other
if boolean, 0 or 1, it works the same as absolute,
if > 1, select the right mode (CarbonInterface::DIFF_RELATIVE_TO_NOW / CarbonInterface::DIFF_RELATIVE_TO_ABSOLUTE)
if array => use named options mode

And I will add ago() in Carbon 2 in a backward-compatible way also using parameter-type detection.

I will add there a comment about the $syntax parameter. But the important thing is, you still can continue to use it as if it just was $absolute boolean flag.

Reply:
You can have a look at:
https://github.com/briannesbitt/Carbon/pull/1654

Here is how I plan to implement jessengers equivalent features, as you can see all unit tests still pass, only differences are some translations divergences.